var/spack/repos/builtin/packages/xtb/package.py

```diff
@@ -23,9 +23,11 @@ class Xtb(MesonPackage):
     depends_on("meson@0.57.2:", type="build")
     depends_on("mctc-lib")
     depends_on("pkgconfig", type="build")
-    depends_on("tblite", when="@6.6.0:")
+    depends_on("tblite", when="+tblite")
+    depends_on("test-drive", type="build")
```

Comment:
```
1. xtb: unknown package 'test-drive' in 'depends_on' directive
     in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/xtb/package.py
2. xtb: dependency on test-drive cannot be satisfied by known versions of test-drive
    happening in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/xtb/package.py
```

Reply:
The patch I posted was not applied completely, the missing bit is

```diff
diff --git a/var/spack/repos/builtin/packages/test-drive/package.py b/var/spack/repos/builtin/packages/test-drive/package.py
new file mode 100644
index 0000000000..92c77f7af5
--- /dev/null
+++ b/var/spack/repos/builtin/packages/test-drive/package.py
@@ -0,0 +1,12 @@
+from spack.package import *
+
+
+class TestDrive(MesonPackage):
+    """Simple testing framework for Fortran packages"""
+
+    homepage = "https://github.com/fortran-lang/test-drive"
+    url = "https://github.com/fortran-lang/test-drive/releases/download/v0.4.0/test-drive-0.4.0.tar.xz"
+
+    maintainers("awvwgk")
+
+    version("0.4.0", "effabe5d46ea937a79f3ea8d37eea43caf38f9f1377398bad0ca02784235e54a")
```

Reply:
sorry, I really dropped the ball here with my previous edits.

So, this PR not only fixes the xtb build but also adds a new test-drive package

I hope the edits are now applied correctly. Apologies again for the delay

Reply:
Previously the test-drive package was downloaded on-demand, however now meson is used with no download option making the build fail. Adding the package as proper dependency is the way to go then.